### PR TITLE
HW1 - Clean Code (linter and github actions)

### PR DESCRIPTION
**Pull request by irynakopyl**
This PR completes the next tasks: 

- added CI jobs to run build, linter and unit tests
- added ESLinter and fixed all found issues

### CODE REVIEW

**Review by Ndm1t — APPROVED** [review]
Everything seems fine)

Not regarding this particular PR, but just in general. I've noticed some callbacks declared right in the function declaration, in my opinion moving those callbacks to separate functions might improve overall readability. As an example subscription router, handler could be declared in a separate file or class and passed to router

**Comment by irynakopyl** [comment]
@Ndm1t Great catch, updated

File: src/controllers/exchanger.controller.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+export class ExchangerController {
+  async getCurrentRate(req: Request, res: Response) {
+    const currentRate = await ExchangerService.getCurrentRate();
+    if (!currentRate) {
+      res.status(StatusCode.Invalid).json('Invalid status value');
+    }
+    res.status(StatusCode.Success).json(currentRate);
+  }
+}
```
Comment (Ndm1t) [inline src/controllers/exchanger.controller.ts]: It is a good approach to export already initialized class if there is no constructor, so it won't be done each time being imported, but it's not a big deal.
e.g.
`export default new ExchangeController()`

**Review by Ndm1t — APPROVED** [review]
Looks great, good job!

File: src/controllers/exchanger.controller.ts (hunk 1 above)
Comment (irynakopyl) [reply src/controllers/exchanger.controller.ts]: Thanks! I decided to change controllers to be a single functions, as there is no need of instances currently

File: src/controllers/subscription.controller.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { Request, Response } from 'express';
+import { validationResult } from 'express-validator';
+import { StatusCode } from '../models/status-codes.model';
+import { SubscriptionsService } from '../services/subscription.service';
+
+export async function subscribe(request: Request, response: Response) {
+  const subscriptionService = new SubscriptionsService();
+  const errors = validationResult(request);
+  if (!errors.isEmpty()) {
+    return response.status(StatusCode.UnprocessableEntity).json(errors.array());
```
Comment (RuslanMelnychenko) [inline src/controllers/subscription.controller.ts]: The use of 422 or 400 error codes is an open question. The 422 error is rarely used. 400 is when there is no required field, the data type is incorrect. 422 is when the value of `email` field is not an email address. In this case, validation checks everything at once, whether the field is present, whether the field is a string, and whether it is an email. In the first two checks, it is more correct to use 400, in the last 422. But more often they do not bother with this and just answer 400.
You will rarely see anyone supporting the 422 code.

This is a comment to think about 🤔
Does not block MR

**Review by RuslanMelnychenko — APPROVED** [review]

File: src/controllers/subscription.controller.ts (hunk 2 above)
Comment (irynakopyl) [reply src/controllers/subscription.controller.ts]: Agreed, and it seems that more general error code would be better for this case. Updated

**Pull request merged**